File: Revisao/incremental6/CartaoDeCredito.java
abstract class CartaoDeCredito {
    protected String nomeTitular;
    protected String numero;
    protected float limite;
    protected float saldo;
    protected Fatura fatura;
    protected boolean ativo;
    protected List<Transacao> historicoTransacoes;

    public CartaoDeCredito(String nomeTitular, String numero, float limite, Fatura fatura) {
        this.nomeTitular = nomeTitular;
        this.numero = numero;
        this.limite = limite;
        this.saldo = 0;
        this.fatura = fatura;
        this.ativo = true;
        this.historicoTransacoes = new ArrayList<>();
    }

    public String getNomeTitular() {
        return nomeTitular;
    }

    public String getNumero() {
        return numero;
    }

    public float getLimite() {
        return limite;
    }

    public float getSaldo() {
        return saldo;
    }

    public Fatura getFatura() {
        return fatura;
    }

    public void adicionarTransacao(Transacao transacao) {
        historicoTransacoes.add(transacao);
        fatura.adicionarTransacao(transacao);
    }

    public void desativarCartao() {
        ativo = false;
        historicoTransacoes.clear();
        fatura.limparFatura();
    }

    // Método que será sobrescrito nas subclasses
    public abstract void realizarCompra(float valor);
}

File: Revisao/python/cartaoDeCredito.py

class CartaoDeCredito:
    def __init__(self, nome_titular, numero, limite, saldo, fatura, ativo, historico_transa):
        self.nome_titular = nome_titular
        self.numero = numero
        self.limite = limite
        self.saldo = saldo
        self.fatura = fatura
        self.ativo = ativo
        self.historico_transa = []
        
    def adicionar_transacao(self, transacao):
        self.historico_transa.append(transacao)
        self.fatura.adicionar_transacao(transacao)
        
    def desativar_cartao(self):
        self.ativo = False
        self.historico_transa.clear()
        self.fatura.limpar_fatura()
        
    